src/gtc/cuir/extent_analysis.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+#
+# GTC Toolchain - GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from collections import defaultdict
+from typing import Any, Dict
+
+from eve import NodeTranslator
+
+from . import cuir
+
+
+class ComputeExtents(NodeTranslator):
```

Comment:
Is this cuir specific? Or will other backends require the same analysis? DaCe?

Reply:
Please add some tests.

Reply:
This works on the CUIR-level, so currently it is indeed very backend-specific. Depending on the strategy and decisions around the new execution model etc., it might make sense to add this or similar information to the OIR. As also the OIR optimizations could be affected, I don’t think that this step should be considered now.